src/tequila/quantumchemistry/orbital_optimizer.py

```diff
@@ -37,7 +39,9 @@ def __call__(self, local_data, *args, **kwargs):
         self.iterations += 1
 
 def optimize_orbitals(molecule, circuit=None, vqe_solver=None, pyscf_arguments=None, silent=False,
-                      vqe_solver_arguments=None, initial_guess=None, return_mcscf=False, use_hcb=False, molecule_factory=None, molecule_arguments=None, *args, **kwargs):
+                      vqe_solver_arguments=None, initial_guess=None, return_mcscf=False, use_hcb=False, 
```

Comment:
if `use_hcb` and `oo_between_spaces` is both set to True we will get errors.
See below.

Similar:
if `oo_between_spaces` is set to True then the routine here implicitly assumes that JordanWigner with "up,down,up,down..." ordering is used as encoding. This is fine, but you should verify this in the beginning.

e.g. via
`mol.transformation.up_then_down` (bool, should be False)
`mol.transformation.name` (str, should be "JordanWigner")